include/audit_events.hpp

```diff
@@ -188,24 +188,21 @@ inline void auditEvent(const crow::Request& req, const std::string& userName,
 {
     std::string opPath;
     char cnfgBuff[256];
-    size_t bufLeft = 256; // Amount left available in cnfgBuff
-    char* user = NULL;
-    size_t opPathLen;
-    size_t userLen = 0;
+    size_t bufLeft = sizeof(cnfgBuff); // Amount left available in cnfgBuff
     int rc;
     int origErrno;
     std::string ipAddress;
-    std::string detail = "";
+    std::string detail;
 
```

Comment:
Itis generally better to define the variables close to where they are needed & with the proper initialization.

For example, rc & origErrno are better be defined with the initial values.
Or those variables are not needed to define if `auditOpen()' is false.

So I would move like

```
inline void auditEvent(const crow::Request& req, const std::string& userName,
                       bool success)
{
    if (!auditOpen())
    {
        return;
    }

==> Declare opPath & buff
    char cnfgBuff[256];
    size_t bufLeft = sizeof(cnfgBuff); // Amount left available in cnfgBuff
   std::string  opPath = "op=" + std::string(req.methodString()) + ":" +
             std::string(req.target()) + " ";
...

==> 'detail' seems needed here
    // Determine any additional info for the event
   std::string detail;
    if (wantDetail(req))
    {
        detail = req.body + " ";
    }

...
==> ipAddress are needed here
   std::string     ipAddress = req.ipAddress.to_string();
....
==> 'rc' is to handle audit_log_...()
   int     rc = audit_log_user_message(auditfd, AUDIT_USYS_CONFIG, cnfgBuff,
                                boost::asio::ip::host_name().c_str(),
                                ipAddress.c_str(), NULL, int(success));
    if (rc <= 0)
    {
        // Something failed with existing connection. Try to establish a new
        // connection and retry if successful.
        // Preserve original errno to report if the retry fails.
==> origErrno is a temporary variable to save the current  errno
        int origErrno = errno;
.....
        if (rc <= 0)
        {
            BMCWEB_LOG_ERROR << "Error writing audit message: " << origErrno;
        }
    }

    return;
}
```

Reply:
That makes sense, I've made these changes.